src/main/java/io/strimzi/kafka/quotas/throttle/PolicyBasedThrottle.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2020, Red Hat Inc.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.quotas.throttle;
+
+import io.strimzi.kafka.quotas.VolumeObserver;
+import io.strimzi.kafka.quotas.VolumeUsageResult;
+import io.strimzi.kafka.quotas.throttle.fallback.ExpiryPolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Clock;
+import java.util.Objects;
+import java.util.function.Function;
+
+import static io.strimzi.kafka.quotas.VolumeUsageResult.VolumeSourceObservationStatus.SUCCESS;
+
+
+/**
+ * Converts volume usage to a throttle factor by applying a {@link ThrottleFactorPolicy} to the observations.
+ * Notifies a listener when the factor changes.
+ */
+public class PolicyBasedThrottle implements VolumeObserver, ThrottleFactorSource {
+    private static final Logger log = LoggerFactory.getLogger(PolicyBasedThrottle.class);
+    private final ThrottleFactorPolicy factorPolicy;
+    private final ExpiryPolicy expiryPolicy;
+    private final Runnable listener;
+    private final Clock clock;
+    private volatile ThrottleFactor throttleFactor;
+    private final double fallbackThrottleFactor;
+
+    /**
+     * @param factorPolicy Which policy to apply
+     * @param listener     the lister to be notified of changes
+     * @param expiryPolicy expiry policy to control how long a factor is applied for
+     * @param fallbackThrottleFactor throttle factor to apply if a factor reaches its expiry
+     */
+    public PolicyBasedThrottle(ThrottleFactorPolicy factorPolicy, Runnable listener, ExpiryPolicy expiryPolicy, double fallbackThrottleFactor) {
+        this(factorPolicy, listener, Clock.systemUTC(), expiryPolicy, fallbackThrottleFactor);
+    }
+
+    /* test */ PolicyBasedThrottle(ThrottleFactorPolicy factorPolicy, Runnable listener, Clock clock, ExpiryPolicy expiryPolicy, double fallbackThrottleFactor) {
+        this.factorPolicy = factorPolicy;
+        this.listener = listener;
+        this.clock = clock;
+        this.expiryPolicy = expiryPolicy;
+        throttleFactor = ThrottleFactor.validFactor(1.0d, clock.instant(), this.expiryPolicy);
+        this.fallbackThrottleFactor = fallbackThrottleFactor;
+    }
+
+    @Override
+    public ThrottleFactor currentThrottleFactor() {
+        return throttleFactor;
+    }
+
+    @Override
+    public void observeVolumeUsage(VolumeUsageResult observedVolumes) {
+        updateFactor(current -> getNewFactor(observedVolumes, current));
+    }
+
+    /**
+     * Check for expired factor and fallback if it has expired.
+     */
+    public void checkThrottleFactorValidity() {
+        log.info("checking for expired factor");
```

Comment:
Log messages typically start with a capital letter. There's a few more instances below and in other files.

Reply:
Fixed in https://github.com/strimzi/kafka-quotas-plugin/pull/37/commits/bcef9fa527f7fad9194558ee442b00b5cb22d248